app/jobs/state_file/import_from_direct_file_job.rb

```diff
@@ -40,11 +40,39 @@ def perform(authorization_code:, intake:)
         intake.df_data_import_errors << DfDataImportError.new(message: err.to_s)
       end
 
+      # removing duplicate associations here because sometimes we create duplicate records during data import
+      # future work will prevent this issue from happening and this can be removed
+      remove_duplicate_w2s(intake)
+      remove_duplicate_dependents(intake)
+
       DfDataTransferJobChannel.broadcast_job_complete(intake)
     end
 
     def priority
       PRIORITY_LOW
     end
+
+    private
+
+    def remove_duplicates(intake, collection, identifying_attribute_name)
```

Comment:
i did not want to send in the intake, but both `StateFileDependent` and `StateFileW2` have different names for the association, so I just pass it in